streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -255,6 +259,13 @@ public Subscription subscription(Set<String> topics) {
             state.capacity = state.capacity + 1d;
         }
 
+        // update the topic groups with the returned subscription list for regex pattern subscriptions
+        if (streamThread.builder.sourceTopicPattern() != null) {
+            SubscriptionUpdates subscriptionUpdates = streamThread.builder.getSubscriptionUpdates();
+            subscriptionUpdates.updateTopics(regexSubscribedTopics);
+            this.topicGroups = streamThread.builder.topicGroups(streamThread.applicationId);
```

Comment:
Since now we will update the `this.topicGroups` if there is a sourceTopicPattern, I'm thinking if now we could remove `this.topicGroups = streamThread.builder.topicGroups(streamThread.applicationId);` in the `configure` function of `StreamPartitionAssignor` (line 121)?